core/src/main/java/org/mule/DefaultMuleMessage.java

```diff
@@ -1322,6 +1353,24 @@ private void writeObject(ObjectOutputStream out) throws Exception
         return toWrite;
     }
 
+    @Override
+    protected Serializable serializeValue(ObjectOutputStream out) throws Exception
+    {
+        byte[] serializablePayload = muleContext.getTransformationService().getPayloadAsBytes(this);
```

Comment:
It's used in TypedValue.writeObject(). It is only performed when payload is not serializable.